File: Dev/C++/Mascaret/src/VEHA/Kernel/LiteralInteger.cpp
#include <cstdio>
#include "VEHA/Kernel/LiteralInteger.h"
#include "VEHA/Kernel/PrimitiveType.h"
#include "MascaretApplication.h"
#include "VEHA/Kernel/Model.h"
#include <boost/lexical_cast.hpp>

namespace VEHA
{

LiteralInteger::LiteralInteger(const int& value)
:LiteralSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("integer")),_value(value)
{
}
LiteralInteger::LiteralInteger(const string & str)
:LiteralSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("integer")),_value( boost::lexical_cast<int>(str))
{
}

LiteralInteger::~LiteralInteger()
{
}

bool LiteralInteger::setValueFromString(const string& value)
{
	try
	{
		_value= boost::lexical_cast<int>(value);
		return true;
	}
	catch(boost::bad_lexical_cast &)
	{
		return false;
	}

	sscanf(value.c_str(),"%d",&_value);
	return true;
}

string LiteralInteger::getStringFromValue() const
{
	return boost::lexical_cast<std::string>(_value);
}

LiteralInteger::operator string () const
		{
	return boost::lexical_cast<std::string>(_value);
		}

LiteralInteger::operator int () const
		{
	return _value;
		}
//const ValueSpecification& LiteralInteger::operator= (int  value)
//{
//	_value=value;
//	return *this;
//}
shared_ptr<ValueSpecification> LiteralInteger::clone() const
{
	LiteralInteger * vs=new LiteralInteger();
	vs->CLASSTYPE=CLASSTYPE;
	vs->_type=_type;
	vs->_value=_value;
	return shared_ptr<ValueSpecification>(vs);
}
//const ValueSpecification& LiteralInteger::operator= (const ValueSpecification&vs)
//{
//	_value=(int)vs;
//	return *this;
//}
}

File: Dev/CS/Mascaret/Mascaret/VEHA/Kernel/LiteralInteger.cs
using System;
using System.Collections.Generic;

namespace Mascaret
{
    public class LiteralInteger : LiteralSpecification
    {

        protected int iValue;
        public int IValue
        {
            get { return iValue; }
            set { iValue = value; }
        }

        public bool setValueFromString(string intvalue)
        {
            iValue = int.Parse(intvalue);
            return true;
        }

        public override string getStringFromValue()
        {
            return (iValue.ToString());
        }

        public override int getIntFromValue()
        {
            return iValue;
        }

        public LiteralInteger(string intvalue)
            : base(MascaretApplication.Instance.Model.getBasicType("integer"))
        {
            setValueFromString(intvalue);
        }

        public LiteralInteger(int i)
            : base(MascaretApplication.Instance.Model.getBasicType("integer"))
        {
            iValue = i;
        }

        public LiteralInteger()
            : base(MascaretApplication.Instance.Model.getBasicType("integer"))
        {
        }

        public override ValueSpecification clone()
        {
            LiteralInteger vs = new LiteralInteger(0);
            vs.CLASSTYPE = CLASSTYPE;
            vs.Type = this.Type;
            vs.iValue = iValue;
            return ((ValueSpecification)vs);
        }

    }
}

